core/class-admin.php

```diff
@@ -903,12 +903,18 @@ public function save_ajax_metabox( $post_id ) {
             }
           }
 
-          return $this->shipment->create_shipment($order, $service_id, $additional_services);
+          $creating_shipment = get_post_meta($post_id, '_' . $this->core->prefix . '_creating_shipment', 'true');
+          if ( empty($creating_shipment) ) {
+            update_post_meta($post_id, '_' . $this->core->prefix . '_creating_shipment', 'true');
+            return $this->shipment->create_shipment($order, $service_id, $additional_services);
```

Comment:
If creating shipment fails, then update_post_meta is never released so it will prevent trying to create shipment again.